extensions/wikia/MercuryApi/MercuryApiController.class.php

```diff
@@ -437,39 +451,12 @@ public function getArticle() {
 				throw new NotFoundApiException( 'Article is empty' );
 			}
 
-			$data[ 'details' ] = $this->getArticleDetails( $article );
-
-			$isMainPage = $title->isMainPage();
-			$data[ 'isMainPage' ] = $isMainPage;
-
-			$titleBuilder = new WikiaHtmlTitle();
-
-			if ( $isMainPage && !empty( $wgEnableMainPageDataMercuryApi ) &&
-				 ( new CommunityDataService( $wgCityId ) )->hasData()
-			) {
-				$data[ 'mainPageData' ] = $this->getMainPageData();
-			} else {
-				$articleAsJson = $this->getArticleJson( $articleId, $sections );
-				$data[ 'article' ] = $articleAsJson;
-				$data[ 'topContributors' ] = $this->getTopContributorsDetails(
-					$this->getTopContributorsPerArticle( $articleId )
-				);
-				$relatedPages = $this->getRelatedPages( $articleId );
-
-				if ( !empty( $relatedPages ) ) {
-					$data[ 'relatedPages' ] = $relatedPages;
-				}
-				if ( !$isMainPage ) {
-					$titleBuilder->setParts( [ $articleAsJson[ 'displayTitle' ] ] );
-				}
-			}
-			$data[ 'htmlTitle' ] = $titleBuilder->getTitle();
-
+			$data['htmlTitle'] = $titleBuilder->getTitle();
```

Comment:
It has to be before the `if` containing `$titleBuilder->setParts`. Moreover, `$data['htmlTitle'] = $titleBuilder->getTitle();` has to be placed after all of that.